### PR TITLE
chore: Remove manual memory management from stats

### PR DESCRIPTION
**Pull request by dranikpg**
Cleans up #2328

### CODE REVIEW

File: src/server/server_state.cc [hunk 1]
```diff
@@ -23,32 +23,12 @@ namespace dfly {
...
-  this->tx_width_freq_arr = other.tx_width_freq_arr;
-  other.tx_width_freq_arr = nullptr;
-
-  return *this;
-}
-
-ServerState::Stats& ServerState::Stats::Add(unsigned num_shards, const ServerState::Stats& other) {
-  static_assert(sizeof(Stats) == 13 * 8, "Stats size mismatch");
+ServerState::Stats& ServerState::Stats::Add(const ServerState::Stats& other) {
+  static_assert(sizeof(Stats) == 14 * 8, "Stats size mismatch");
```
Comment (BorysTheDev) [inline src/server/server_state.cc:31]: I think this assertion is better move under the Stats definition

Comment (BorysTheDev) [reply src/server/server_state.cc:31]: and instead of 8 use sizeof(int64_t)

File: src/server/server_state.cc [hunk 2]
```diff
@@ -65,15 +45,12 @@ ServerState::Stats& ServerState::Stats::Add(unsigned num_shards, const ServerSta
...
-    this->tx_width_freq_arr = new uint64_t[num_shards];
-    std::copy_n(other.tx_width_freq_arr, num_shards, this->tx_width_freq_arr);
+  if (this->tx_width_freq_arr.size() > 0) {
+    DCHECK_EQ(this->tx_width_freq_arr.size(), other.tx_width_freq_arr.size());
+    this->tx_width_freq_arr += other.tx_width_freq_arr;
   } else {
-    for (unsigned i = 0; i < num_shards; ++i) {
-      this->tx_width_freq_arr[i] += other.tx_width_freq_arr[i];
-    }
+    this->tx_width_freq_arr = other.tx_width_freq_arr;
```
Comment (BorysTheDev) [inline src/server/server_state.cc:52]: should never happen

**Review by BorysTheDev — APPROVED** [review]

**Comment by dranikpg** [comment]
Accidentally deleted this branch 😬

**Pull request merged**